modules/sdk-core/src/bitgo/wallet/wallets.ts

```diff
@@ -789,7 +789,7 @@ export class Wallets implements IWallets {
     originalPasscodeEncryptionCode,
     backupProvider,
   }: GenerateMpcWalletOptions): Promise<WalletWithKeychains> {
-    if (multisigType === 'tss' && this.baseCoin.getMPCAlgorithm() === 'ecdsa' && walletVersion === 3) {
+    if (multisigType === 'tss' && this.baseCoin.getMPCAlgorithm() === 'ecdsa') {
```

Comment:
in this case why not remove `walletVersion` from the parameters and determine the wallet version based on the tss Settings? Otherwise you're leaving a hole if a user passes in `undefined or 2` and they don't have mpcv2 enabled.

Reply:
because if MPCv2 is disabled, we do nothing and let the flow continues with whatever walletVersion was passed in

Reply:
is wallet version 5 blocked if the enterprise/user doesn't have the correct flags?

Reply:
yes, depends on the user flag or the env flag, only if one of those is enabled it works